code/modules/mob/living/simple_animal/friendly/corgi.dm

```diff
@@ -529,4 +529,38 @@
 				emote("yaps happily")
 		else
 			if(M && stat != DEAD) // Same check here, even though emote checks it as well (poor form to check it only in the help case)
-				emote("growls")
\ No newline at end of file
+				emote("growls")
+
+
+//Sasha isn't even a corgi you dummy!
+/mob/living/simple_animal/corgi/Sasha
+	name = "Sasha"
+	real_name = "Sasha"
+	gender = FEMALE
+	desc = "It's a doberman, how intimidating!"
+	icon_state = "doby"
+	icon_living = "doby"
+	icon_dead = "doby_dead"
+	response_help  = "pets"
+	response_disarm = "bops"
+	response_harm   = "kicks"
+	var/turns_since_scan = 0
+
+//Sasha can't wear hats!
+/mob/living/simple_animal/corgi/Sasha/Topic(href, href_list)
+	if(href_list["remove_inv"] || href_list["add_inv"])
+		usr << "<span class='warning'>[src] won't wear that!</span>"
+		return
+	..()
+
+/mob/living/simple_animal/corgi/Sasha/Life()
+	..()
+
+	if(!stat && !resting && !buckled)

```

Comment:
But Lisa has a different variation of it, with the whole "She already has a cute bow!" thing, unless you mean specifically the if(!stat && !resting && !buckled) thing